File: kitten-house-care/src/main/java/org/jesperancinha/housing/service/OwnerServiceImpl.java
package org.jesperancinha.housing.service;

import org.jesperancinha.housing.repository.OwnerRepositoryImpl;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class OwnerServiceImpl {

    private final OwnerRepositoryImpl ownerRepository;

    public OwnerServiceImpl(OwnerRepositoryImpl ownerRepository) {
        this.ownerRepository = ownerRepository;
    }

    public Mono<String> checkLiability(String name) {
        return ownerRepository.checkLiability(name);
    }

}

File: kitten-house-care/src/main/kotlin/org/jesperancinha/housing/service/OwnerServiceImpl.kt
package org.jesperancinha.housing.service

import org.jesperancinha.housing.dao.OwnerRepository
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class OwnerServiceImpl(private val ownerRepository: OwnerRepository) {
    fun checkLiability(name: String): Mono<String> {
        return ownerRepository.checkLiability(name)
    }
}